src/main/java/uk/gov/ons/ctp/response/casesvc/service/CaseService.java

```diff
@@ -65,91 +195,411 @@
+              .build();
+      targetCase.getResponses().add(response);
+      caseRepo.save(targetCase);
+    }
+  }
+
+  /**
+   * Effect a state transition for the target case if the category indicates one is required If a
+   * transition was made and the state changes as a result, notify the action service of the state
+   * change AND if the event was type DISABLED then also call the IAC service to disable/deactivate
+   * the IAC code related to the target case.
+   *
+   * @param category the category details of the event
+   * @param targetCase the 'source' case the event is being created for
+   * @throws CTPException when case state transition error
+   */
+  private void effectTargetCaseStateTransition(Category category, Case targetCase)
+      throws CTPException {
+    CaseDTO.CaseEvent transitionEvent = category.getEventType();
+    if (transitionEvent != null) {
+      if ((transitionEvent == CaseDTO.CaseEvent.DISABLED
+              && !category
+                  .getCategoryName()
+                  .equals(CategoryDTO.CategoryName.SUCCESSFUL_RESPONSE_UPLOAD))
+          || transitionEvent == CaseDTO.CaseEvent.ACTIONPLAN_CHANGED) {
+        if (targetCase.getIac() != null) {
+          internetAccessCodeSvcClient.disableIAC(targetCase.getIac());
+        }
+      }
+      CaseState oldState = targetCase.getState();
+      CaseState newState = caseSvcStateTransitionManager.transition(oldState, transitionEvent);
+
+      if (!oldState.equals(newState)) {
+        targetCase.setState(newState);
+        caseRepo.saveAndFlush(targetCase);
+        notificationPublisher.sendNotification(
+            prepareCaseNotification(targetCase, transitionEvent));
+      }
+    }
+  }
+
+  /**
+   * Create an event for a newly created case
+   *
+   * @param caze the case for which we want to record the event
+   * @param caseEventCategory the category of the event that led to the creation of the case
+   */
+  private void createCaseCreatedEvent(Case caze, Category caseEventCategory) {
+    CaseEvent newCaseCaseEvent = new CaseEvent();
+    newCaseCaseEvent.setCaseFK(caze.getCasePK());
+    newCaseCaseEvent.setCategory(CategoryDTO.CategoryName.CASE_CREATED);
+    newCaseCaseEvent.setCreatedBy(Constants.SYSTEM);
+    newCaseCaseEvent.setCreatedDateTime(DateTimeUtil.nowUTC());
+    newCaseCaseEvent.setDescription(
+        String.format(CASE_CREATED_EVENT_DESCRIPTION, caseEventCategory.getShortDescription()));
+
+    caseEventRepo.saveAndFlush(newCaseCaseEvent);
+  }
+
+  /**
+   * Create the CaseGroup for the Case.
+   *
+   * @param caseGroupData SampleUnitParent from which to create CaseGroup.
+   * @return newcaseGroup created caseGroup.
+   */
+  private CaseGroup createNewCaseGroup(SampleUnitParent caseGroupData) {
+    CaseGroup newCaseGroup = new CaseGroup();
+
+    newCaseGroup.setId(UUID.randomUUID());
+    if (caseGroupData.getPartyId() != null) {
+      newCaseGroup.setPartyId(UUID.fromString(caseGroupData.getPartyId()));
```

Comment:
we so need to change all the places where String is used instead of UUID... but that's a job for another PR